audit/base/base.go

```diff
@@ -98,6 +101,26 @@ func IsUnrecoverable(err error) bool {
 	return false
 }
 
+type metrics struct {
+	meter        metric.Meter
+	msgsSent     metric.Int64Counter
+	msgsRequeued metric.Int64Counter
+	msgsDropped  metric.Int64Counter
+	msgErrs      metric.Int64Counter
+
+	// requeuedCounter exists to track the number of requeued messages in a test.
+	// You cannot extract the value from an otel counter.
+	requeuedCounter atomic.Uint64
+}
+
+func (m *metrics) init() {
```

Comment:
minor, style: Why not `newMetrics() *metrics`, rather than construct + init? (Not asking you to change just curious your thoughts)

Reply:
Acutally no good reason.  Was copying examples that had an init, though theirs was slightly different.  Just kept the name.  It really should be newMetrics(), so I changed it.